pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/SegmentMetadataImpl.java

```diff
@@ -396,6 +413,26 @@ public String getIndexType() {
   }
 
   @Override
+  public String getTimeColumn() {
+    return _timeColumn;
+  }
+
+  @Override
+  public long getStartTime() {
+    return _segmentStartTime;
```

Comment:
No it cannot be reused, because its semantics does not match start/end of segment metadata. Start and end are both inclusive in segment metadata, whereas for Interval class, the end is exclusive. Besides, it is inefficient because it requires unnecessary conversion to millis.